functions/src/api/v1/group/createGroup.ts

```diff
@@ -47,12 +63,39 @@ export const createGroup = wrap(async (req: Request, res: Response) => {
     await assertNewResourceCreationGranted(userId)
 
     const name = req.body.name
+    const requestedGroupId = req.body.groupId
+    const password = req.body.password
     const members = JSONParse(req.body.members || '[]')
 
-    const groupId = await GroupDao.add(userId, name, members)
-    await GroupDao.update({
-        id: groupId,
-    })
+    if (!password) {
+        throw new BadRequestError('Password is required')
+    }
+
+    let groupId: GroupId
+    if (requestedGroupId) {
+        try {
+            await GroupDao.get(requestedGroupId)
+            throw new GroupConflictError()
+        } catch (error) {
+            if (error instanceof GroupNotFoundError) {
+                await GroupDao.set(
```

Comment:
why `set` ? Shouldn't `create` only do an `add` ?

Reply:
here we want to add a group with a fixed id in advance, requiring set, not add.

Reply:
(non blocking) I think `createGroup` logic may be expressed in a clearer way